modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/sql/SqlDdlParserTest.java

```diff
@@ -0,0 +1,299 @@
+        assertThat(node, instanceOf(IgniteSqlCreateTable.class));
+
+        IgniteSqlCreateTable createTable = (IgniteSqlCreateTable)node;
+
+        assertThat(createTable.name().names, is(ImmutableList.of("MY_TABLE")));
+        assertThat(createTable.ifNotExists, is(true));
+        assertThat(createTable.columnList(), hasItem(columnWithName("ID")));
+        assertThat(createTable.columnList(), hasItem(columnWithName("VAL")));
+    }
+
+    /**
+     * Parsing of CREATE TABLE with specified PK constraint where constraint
+     * is a shortcut within a column definition.
+     */
+    @Test
+    public void createTableWithPkCase1() throws SqlParseException {
+        String query = "create table my_table(id int primary key, val varchar)";
+
+        SqlNode node = parse(query);
+
+        assertThat(node, instanceOf(IgniteSqlCreateTable.class));
+
+        IgniteSqlCreateTable createTable = (IgniteSqlCreateTable)node;
+
+        assertThat(createTable.name().names, is(ImmutableList.of("MY_TABLE")));
+        assertThat(createTable.ifNotExists, is(false));
+        assertThat(createTable.columnList(), hasItem(ofTypeMatching(
+            "PK constraint with name \"ID\"", SqlKeyConstraint.class,
+            constraint -> hasItem(ofTypeMatching("identifier \"ID\"", SqlIdentifier.class, id -> "ID".equals(id.names.get(0))))
+                .matches(constraint.getOperandList().get(1))
+                && constraint.getOperandList().get(0) == null
+                && constraint.isA(singleton(SqlKind.PRIMARY_KEY)))));
+    }
+
+    /**
+     * Parsing of CREATE TABLE with specified PK constraint where constraint
+     * is set explicitly and has no name.
+     */
+    @Test
+    public void createTableWithPkCase2() throws SqlParseException {
+        String query = "create table my_table(id int, val varchar, primary key(id))";
+
+        SqlNode node = parse(query);
+
+        assertThat(node, instanceOf(IgniteSqlCreateTable.class));
+
+        IgniteSqlCreateTable createTable = (IgniteSqlCreateTable)node;
+
+        assertThat(createTable.name().names, is(ImmutableList.of("MY_TABLE")));
+        assertThat(createTable.ifNotExists, is(false));
+        assertThat(createTable.columnList(), hasItem(ofTypeMatching(
+            "PK constraint without name containing column \"ID\"", SqlKeyConstraint.class,
+            constraint -> hasItem(ofTypeMatching("identifier \"ID\"", SqlIdentifier.class, id -> "ID".equals(id.names.get(0))))
+                .matches(constraint.getOperandList().get(1))
+                && constraint.getOperandList().get(0) == null
+                && constraint.isA(singleton(SqlKind.PRIMARY_KEY)))));
+    }
+
+    /**
+     * Parsing of CREATE TABLE with specified PK constraint where constraint
+     * is set explicitly and has a name.
+     */
+    @Test
+    public void createTableWithPkCase3() throws SqlParseException {
+        String query = "create table my_table(id int, val varchar, constraint pk_key primary key(id))";
```

Comment:
can u also append test for multiple pk in constraint ? i.e:
primary key(id1, id2)

Reply:
done